### PR TITLE
refactor(examples): Update react example for new component architecture

### PR DESCRIPTION
**Pull request by traviskaufman**
* Refactor react components to use correct classes, new DOM structure, and
  foundations/adapters
* Update examples README documenting that only angular2 isn't
  functional
* Remove checkbox mixin file
* Export MDLCheckboxFoundation from checkbox (eek!)

Part of #4569 (cc #4578)
[#126819269]

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
-# NOTE: Not currently functional.
+# MDL Framework Integration Examples
 
-We've recently [changed our architecture]() such that these examples are outdated and probably don't
-work. We will fix them up before our beta release, as things may still change until then.
+> NOTE: Angular2 example currently not function. Working angular2 example can be found on our [POC branch](https://github.com/google/material-design-lite/tree/experimental/v2-architecture-poc/examples/angular2).
```
Comment (sgomes) [inline examples/README.md]: s/function/functional/

File: examples/README.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
 
-Working framework examples for v2 can be found on our [POC branch](https://github.com/google/material-design-lite/tree/experimental/v2-architecture-poc/examples).
+This directory contains examples show how you can easily integrate MDL components into any framework
```
Comment (sgomes) [inline examples/README.md]: examples that show / examples which show

**Comment by sgomes** [comment]
LGTM other than the minor wording nits I pointed out. Thanks for updating the example, @traviskaufman!

**Comment by traviskaufman** [comment]
Merging before CI lands as update was docs only and CI had passed beforehand.

**Pull request merged**